streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorNode.java

```diff
@@ -98,4 +116,47 @@ public String toString() {
         }
         return sb.toString();
     }
+
+    protected class NodeMetrics  {
+        final StreamsMetrics metrics;
+        final String metricGrpName;
+        final Map<String, String> metricTags;
+
+        final Sensor nodeProcessTimeSensor;
+        final Sensor nodePunctuateTimeSensor;
+        final Sensor contextForwardSensor;
+        final Sensor nodeTaskCreationSensor;
+        final Sensor nodeTaskDestructionSensor;
+
+
+        public NodeMetrics(StreamsMetrics metrics, String name, String sensorNamePrefix) {
+            String scope = "node";
+            this.metrics = metrics;
+            this.metricGrpName = "node-metrics-" + name;
+            this.metricTags = new LinkedHashMap<>();
+            this.metricTags.put("node-id", "-" + name);
+
+            // these are all latency metrics
+            this.nodeProcessTimeSensor = metrics.addLatencySensor(scope, name, "process", Sensor.RecordLevel.SENSOR_DEBUG);
+            this.nodePunctuateTimeSensor = metrics.addLatencySensor(scope, name, "punctuate", Sensor.RecordLevel.SENSOR_DEBUG);
+
+            this.contextForwardSensor = metrics.sensor(sensorNamePrefix + "node-forward-time" + name, Sensor.RecordLevel.SENSOR_DEBUG);
```

Comment:
@enothereska My suggestion is to remove the suffix of `name`, which is the processor name in sensor, since it is already included in the tags. 

Take the `JmxReporter` as an example, it generates the mbean for streams as:

```
mbean: "kafka-streams": type="streams-processor-node-metrics", "node-id"="PROCESSOR-XXX.."
    attribute1: "task.1-1.processor-throughput-PROCESSOR-XXX"
    attribute2: "task.1-1.processor-create-time-PROCESSOR-XXX"
    attribute3: "task.1-1.processor-destruction-time-PROCESSOR-XXX"
    attribute4: "task.1-2.processor-throughput-PROCESSOR-XXX"
    attribute5: "task.1-2.processor-create-time-PROCESSOR-XXX"
    attribute6: "task.1-2.processor-destruction-time-PROCESSOR-XXX"
    ...
```

where `kafka-streams` is the prefix passed in `KafkaStreams`, and `streams-processor-node-metrics` is the pre-defined `metricGrpName `. You can see here that the processor name suffix in attribute names are redundant, and more generally for any reporters implementations, metrics will be grouped by "group name" and "tags".